### ISSUE TITLE
Support for CAUSAL padding mode in conv1dWithBias is not implemented yet - Is there plans to implement this?

### ISSUE THREAD

**Issue by shilsircar** [tensorflow/tfjs#3578]
To get help from the community, we encourage using Stack Overflow and the [`tensorflow.js`](https://stackoverflow.com/questions/tagged/tensorflow.js) tag.

#### TensorFlow.js version
tfjs@2.0.1
#### Browser version
Chrome m80
#### Describe the problem or feature request
I have converted a Keras model to TFJS as follows:
model_class.model.load_weights('./pretrained_model/model.h5')
tfjs.converters.save_keras_model(./jsmodel/')

Then I have converted the custom layers in the model in TJFS and serialized them. Below is the model summary:

tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 Layer (type)                    Output shape         Param #     Receives inputs                  
tfjs@2.0.1:17 ==================================================================================================
tfjs@2.0.1:17 input_1 (InputLayer)            [1,512]              0                                            
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 ff_tlayer (FFTlayer)            [[1,1,257],[1,1,257] 0           input_1[0][0]                    
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 lstm (LSTM)                     [1,1,128]            197632      ff_tlayer[0][0]                  
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 dropout (Dropout)               [1,1,128]            0           lstm[0][0]                       
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 lstm_1 (LSTM)                   [1,1,128]            131584      dropout[0][0]                    
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 dense (Dense)                   [1,1,257]            33153       lstm_1[0][0]                     
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 activation (Activation)         [1,1,257]            0           dense[0][0]                      
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 multiply (Multiply)             [1,1,257]            0           ff_tlayer[0][0]                  
tfjs@2.0.1:17                                                                  activation[0][0]                 
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 iff_tlayer (IFFTlayer)          [1,1,512]            0           multiply[0][0]                   
tfjs@2.0.1:17                                                                  ff_tlayer[0][1]                  
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 conv1d (Conv1D)                 [1,1,256]            131072      iff_tlayer[0][0]                 
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 instant_layer_normalization (In [1,1,256]            512         conv1d[0][0]                     
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 lstm_2 (LSTM)                   [1,1,128]            197120      instant_layer_normalization[0][0]
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 dropout_1 (Dropout)             [1,1,128]            0           lstm_2[0][0]                     
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 lstm_3 (LSTM)                   [1,1,128]            131584      dropout_1[0][0]                  
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 dense_1 (Dense)                 [1,1,256]            33024       lstm_3[0][0]                     
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 activation_1 (Activation)       [1,1,256]            0           dense_1[0][0]                    
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 multiply_1 (Multiply)           [1,1,256]            0           conv1d[0][0]                     
tfjs@2.0.1:17                                                                  activation_1[0][0]               
tfjs@2.0.1:17 __________________________________________________________________________________________________
tfjs@2.0.1:17 conv1d_1 (Conv1D)               [1,1,512]            131072      multiply_1[0][0]                 
tfjs@2.0.1:17 ==================================================================================================
tfjs@2.0.1:17 Total params: 986753
tfjs@2.0.1:17 Trainable params: 986753
tfjs@2.0.1:17 Non-trainable params: 0
tfjs@2.0.1:17 __________________________________________________________________________________________________

#### Code to reproduce the bug / link to feature request
If you would like to get help from the community, we encourage using Stack Overflow and the [`tensorflow.js`](https://stackoverflow.com/questions/tagged/tensorflow.js) tag.

GitHub issues for this repository are tracked in the [tfjs union repository](https://github.com/tensorflow/tfjs/issues).

Please file your issue there, following the guidance in [that issue template](https://github.com/tensorflow/tfjs/blob/master/ISSUE_TEMPLATE.md).

The issue I am facing is that while trying to run inference on the model. I get this TFJS error:

**Uncaught (in promise) Error: The support for CAUSAL padding mode in conv1dWithBias is not implemented yet.**

is there another way to approach this or is this support in your roadmap anytime soon ?

**Comment by rthadur** [comment]
@shilsircar not any time soon , let us know if you would like to contribute ?

**Comment by shilsircar** [comment]
I might will let u know.

However what I discovered was a bug. My conv1d layer has bias set to false for causal padding. Yet it tries to apply bias for the padding. Is that a known issue ?

**Comment by rthadur** [comment]
@shilsircar please provide the minimal code in order for us to reproduce ?

**Comment by shilsircar** [comment]
[bugreport-conv-causal.zip](https://github.com/tensorflow/tfjs/files/4938247/bugreport-conv-causal.zip)
Ok attached is the sample reference that demonstrates the problem. You have to place the model in a webserver to run the inference. Then change the IP in the provided html where to load the model from. Hopefully this helps.

**Comment by takeraparterer** [comment]
Same problem, even when bias is false.
